lib/asciidoctor/parser.rb

```diff
@@ -1287,25 +1287,26 @@ def self.next_list_item(reader, list_block, match, sibling_trait = nil, style =
       has_text = true
       list_item = ListItem.new(list_block, (item_text = match[2]))
       list_item.source_location = reader.cursor if list_block.document.sourcemap
+      if item_text.start_with?('[')
```

Comment:
A lot of this code only applies to :ulist, so moving it outside of the conditional block will cause it to be evaluated too often and even used in cases it shouldn't. Instead, the change should be as follows:

1. Find elsif list_type == :olist
2. Add these lines below it:

```ruby
if item_text.start_with?('[[') && LeadingInlineAnchorRx =~ item_text
  catalog_inline_anchor $1, $2, list_item, reader
end
```

We'll deal with :colist in a separate issue since I was intentionally not supporting that case.

Reply:
Okay, I guess it is quite simple then in that case. I have updated the pull request accordingly. Thanks!